dxlclient/client.py

```diff
@@ -1255,3 +1235,51 @@ def unregister_service_sync(self, service_req_info, timeout):
 
             self._service_manager.remove_service(service_req_info.service_id)
             service_req_info._wait_for_unregistration(timeout=timeout)
+
+    def _get_mqtt_client(self):
+        """
+        Returns the mqtt client instance based on if web socket proxy is enabled or not
+        :return: MQTT client instance
+        """
+        if self._proxy is not None:
+            mqtt_client = mqtt_dxl
+        else:
+            mqtt_client = mqtt
+
+        return mqtt_client.Client(client_id=self._config._client_id,
+                                  clean_session=True,
+                                  userdata=self,
+                                  protocol=mqtt.MQTTv31,
+                                  transport="websockets" if self.config.use_websockets else "tcp")
+
+    def connect_to_broker(self, broker, keep_alive_interval, ip_or_hostname):
+        """
+        Connect using broker ip or hostname
+        :param broker: Broker reference
+        :param keep_alive_interval: Keep alive interval
+        :param ip_or_hostname: Broker ip or hostname
+        """
+        logger.info("Trying to connect to broker %s...", broker.to_string())
+        # If HTTP proxy for websocket is present, connect using proxy
+        if self._proxy is not None:
```

Comment:
This if block is about whether self._proxy has a value but it is also more about the type of self._client since one of the possible types of self._client does not support the fourth parameter of "self._proxy". This is a consequence of using both the released Paho MQTT client and our forked Paho MQTT client. See lines 19 and 20 of this class for the import statements and the _get_mqtt_client method. Having both libraries seems problematic me from a Java developer's perspective and we could simplify this code by just having one Paho MQTT client which has one connect method which decides if it needs to use a proxy or not instead of having to do that logic here. However, I am not familiar enough with Python to know if this is ok practice and perhaps using two of the same libraries is ok.

Reply:
I agree. We can remove the normal paho-mqtt from 'install_requires' and only keep the paho-mqtt forked version.